tests/test_client.py

```diff
@@ -0,0 +1,373 @@
+
+
+    def test_addparam_period(self):
+        sample_dict = {'test': None}
+
+        test_period = "Test Period"
+
+        VirusTotalApiClient._add_period_param(
+            sample_dict,
+            test_period
+        )
+        self.assertEqual(sample_dict[VirusTotalApiClient._PARAM_PERIOD], test_period)
+
+
+    def test_addparam_repeat(self):
+        sample_dict = {'test': None}
+
+        test_repeat = "Test Repeat"
+
+        VirusTotalApiClient._add_repeat_param(
+            sample_dict,
+            test_repeat
+        )
+        self.assertEqual(sample_dict[VirusTotalApiClient._PARAM_REPEAT], test_repeat)
+
+
+    def test_addparam_notifyurl(self):
+        sample_dict = {'test': None}
+
+        test_notify_url = "http://www.notifyurl.com"
+
+        VirusTotalApiClient._add_notify_url_param(
+            sample_dict,
+            test_notify_url
+        )
+        self.assertEqual(sample_dict[VirusTotalApiClient._PARAM_NOTIFY_URL], test_notify_url)
+
+
+    def test_addparam_scan(self):
+        sample_dict = {'test': None}
+
+        VirusTotalApiClient._add_scan_param(
+            sample_dict,
+            "Anything",
+        )
+        self.assertTrue(sample_dict[VirusTotalApiClient._PARAM_SCAN])
+
+
+    def test_addparam_domain(self):
+        sample_dict = {'test': None}
+
+        test_domain = "testdomain.com"
+
+        VirusTotalApiClient._add_domain_param(
+            sample_dict,
+            test_domain
+        )
+        self.assertEqual(sample_dict[VirusTotalApiClient._PARAM_DOMAIN], test_domain)
+
+
+    def test_addparambyname_date(self):
+        sample_dict = {'date': None}
+        sample_date = SAMPLE_DATE
+        VirusTotalApiClient._add_date_param_by_name(sample_dict, 'date', sample_date)
+
+        self.assertEqual(
+            sample_dict['date'],
+            sample_date.strftime("%Y%m%d%H%M%S")
+        )
+
+
+    def test_addparambyname_bool_dne(self):
+        sample_dict = {'test': None}
+        VirusTotalApiClient._add_boolean_param_by_name(sample_dict, 'does_not_exist', False)
+        with self.assertRaises(KeyError):
+            print(sample_dict['does_not_exist']) #pylint: disable=superfluous-parens
+
+
+    def test_addparambyname_boolfalse(self):
+        sample_dict = {'test': None}
+        VirusTotalApiClient._add_boolean_param_by_name(sample_dict, 'test', False)
+        self.assertIsNone(sample_dict['test'])
+
+
+    def test_addparambyname_bool(self):
+        sample_dict = {'test': None}
+        VirusTotalApiClient._add_boolean_param_by_name(sample_dict, 'test', True)
+        self.assertEqual(sample_dict['test'], "1")
+
+
+    def test_addparambyname_string_dne(self):
+        sample_dict = {'test': None}
+        VirusTotalApiClient._add_string_param_by_name(sample_dict, 'does_not_exist', None)
+        with self.assertRaises(KeyError):
+            print(sample_dict['does_not_exist']) #pylint: disable=superfluous-parens
```

Comment:
Why does this need to be in a `print` statement?

fyi, I think you can avoid the need to have to suppress the `superfluous-parens` pylint error by putting the following line at the top of the module:

```python
from __future__ import print_function
```

This should allow the code to be run-time compiled for Python 2 using the Python 3 behavior, where parens are always required when calling `print`.

Reply:
I just wanted a simple way to force Python to try to reference a key which should not be present in the dictionary without something like an assignment. However, I can change it to simply be a reference statement that does nothing other than try to read the key (and tell pylint to ignore it).